.scalafmt.conf

```diff
@@ -1,23 +1,54 @@
-align = true
+version = 3.3.1
+runner.dialect = Scala213Source3
+
+align.preset = true
 assumeStandardLibraryStripMargin = true
-danglingParentheses = true
-docstrings = JavaDoc
+
+danglingParentheses.preset = true
+
+docstrings.style = Asterisk
+docstrings.wrap = no
+
 maxColumn = 120
-#scalafmt does not support Scala 3 syntax yet (`inline`)
-project.excludeFilters += scala-3
 project.git = true
-rewrite.rules = [ AvoidInfix, ExpandImportSelectors, RedundantParens, SortModifiers, PreferCurlyFors ]
+
+rewrite.rules = [
+  AvoidInfix,
+  RedundantParens,
+  SortModifiers,
+  PreferCurlyFors
+]
 rewrite.neverInfix.excludeFilters = [
   # better for play-json dsl
   and, andKeep, andThen,
   # For scalatest
   in, should, when, must mustEqual, mustBe, "must_==="
 ]
-rewrite.sortModifiers.order = [ "private", "protected", "final", "sealed", "abstract", "implicit", "override", "lazy" ]
+rewrite.sortModifiers.order = [
+  "private",
+  "protected",
+  "final",
+  "sealed",
+  "abstract",
+  "implicit",
+  "override",
+  "lazy"
+]
+
 spaces.inImportCurlyBraces = true   # more idiomatic to include whitespace in import x.{ yyy }
 trailingCommas = preserve
-version = 2.3.2
 
 literals.long=Upper
 literals.float=Upper
 literals.double=Upper
+literals.scientific=Upper
+
+fileOverride {
+  "glob:**/*/src/*/scala-3/**/*.scala" {
+    runner.dialect = scala3
+  }
+}
+
+project.excludePaths = [
+  "glob:**/docs/**/code*/*.scala"
```

Comment:
Make sure special comment to extract snipped are unchanged (special comment a doc delimiters)